src/main/java/org/assertj/core/error/ShouldHaveBinaryContent.java

```diff
@@ -45,13 +46,28 @@ public static ErrorMessageFactory shouldHaveBinaryContent(Path actual, BinaryDif
     return new ShouldHaveBinaryContent(actual, diff);
   }
 
+  /**
+   * Creates a new <code>{@link ShouldHaveBinaryContent}</code>.
+   * @param actual the actual input stream in the failed assertion.
+   * @param diff the differences between {@code actual} and the given binary content.
+   * @return the created {@code ErrorMessageFactory}.
+   */
+  public static ErrorMessageFactory shouldHaveBinaryContent(InputStream actual, BinaryDiffResult diff) {
```

Comment:
I know we did not have tests for the other factory method but it would be nice to have a new test class `ShouldHaveBinaryContent_create_Test` to test that method, you can use `ShouldHaveMessage_create_Test` as an example.